src/axom/quest/io/STEPReader.hpp

```diff
@@ -0,0 +1,126 @@
+  axom::primal::BoundingBox<double, 3> physicalBBox;
+  axom::Array<bool> trimmingCurves_originallyPeriodic;
+};
+
+using PatchDataMap = std::map<int, PatchData>;
+
+}  // namespace internal
+
+/*
+ * \class STEPReader
+ *
+ * \brief A class to help with reading a STEP file containing a parametric BRep (Boundary Representation)
+ * consisting of trimmed NURBS patches.
+ */
+class STEPReader
+{
+public:
+  using NURBSPatch = axom::primal::NURBSPatch<double, 3>;
+  using PatchArray = axom::Array<NURBSPatch>;
+
+  using NURBSCurve = axom::primal::NURBSCurve<double, 2>;
+
+  STEPReader() = default;
+  virtual ~STEPReader();
+
+public:
+  /// Sets the name of the step file to load. Must be called before \a read()
+  void setFileName(const std::string& fileName) { m_fileName = fileName; }
+
+  void setVerbosity(bool verbosity) { m_verbosity = verbosity; }
+
+  /*!
+   * \brief Read the contour file provided by \a setFileName()
+   *
+   * \param[in] validate Adds validation tests on the model, when true
+   * \return 0 for a successful read; non-zero otherwise
+   */
+  virtual int read(bool validate);
+
+  std::string getFileUnits() const;
+  const TopoDS_Shape& getShape() const;
+
+  PatchArray& getPatchArray() { return m_patches; }
+  const PatchArray& getPatchArray() const { return m_patches; }
+
+  /// Logs some information about the loaded BRep
+  void printBRepStats() const;
+
+  /*!
+   * \brief Generates a triangulated representation of the STEP file as a Mint unstructured triangle mesh.
+   *
+   * \param[inout] mesh Pointer to a Mint unstructured mesh that will be populated
+   *            with triangular elements approximating the STEP geometry.
+   * \param[in] linear_deflection Maximum allowed deviation between the
+   *            original geometry and the triangulated approximation.
+   * \param[in] angular_deflection Maximum allowed angular deviation (in radians)
+   *            between normals of adjacent triangles.
+   * \param[in] is_relative When false (default), linear deflection is in mesh units. When true,
+                linear deflection is relative to the local edge length of the triangles.
+   * \param[in] trimmed If true (default), the triangulation respects trimming curves.
+   *            otherwise, we triangulate the untrimmed patches. The latter is mostly to aid 
+   *            in understanding the model's patches and is not generally useful.
+   */
+  virtual int getTriangleMesh(axom::mint::UnstructuredMesh<axom::mint::SINGLE_SHAPE>* mesh,
+                              double linear_deflection = 0.1,
+                              double angular_deflection = 0.5,
+                              bool is_relative = false,
+                              bool trimmed = true);
+
+protected:
+  // open cascade does not appear to offer a direct way to get the number of patches
+  int numPatchesInFile() const;
+
+  const internal::PatchDataMap& getPatchDataMap() const;
```

Comment:
We're only currently using this (and the associated struct at the top of this file) on rank 0, 
so I'm considering moving it to the cpp file.

Reply:
Done.